### DISCUSSION TITLE
Function names not minified

### DISCUSSION THREAD

**Discussion by arwinvdv** [vitejs/vite-plugin-vue#171] [Q&A] [answered]
### Describe the bug

I'm using vite-plugin-vue to build my Vue 3 app and I noticed that the function names are not minified in the production build. This increases the bundle size and exposes some implementation details that I would rather keep hidden. I checked the vite.config.js file and I have minify: 'terser' set as an option, but it doesn't seem to have any effect on the function names. Is this a bug or a limitation of the plugin? Is there any way to enable function name minification with vite-plugin-vue?

I created a example repo, here the changes to recreate: https://github.com/arwinvdv/vue3-template/commit/18c6cb8907581c0a3b5a8d6dc10d48dc00af1196

### Reproduction

https://github.com/arwinvdv/vue3-template

### Steps to reproduce

Checkout this repo: https://github.com/arwinvdv/vue3-template

Run:
```
npm install
npm run build
```
In the created js file the functions `customFunctionThatIsNotMinified` and `anotherCustomFunctionThatIsNotMinified` are visible.

### System Info

```shell
System:
    OS: macOS 13.3.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 66.70 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.9.1 - ~/.nvm/versions/node/v18.9.1/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v18.9.1/bin/yarn
    npm: 8.19.1 - ~/.nvm/versions/node/v18.9.1/bin/npm
  Browsers:
    Chrome: 113.0.5672.63
    Edge: 112.0.1722.68
    Firefox: 112.0.2
    Safari: 16.4
  npmPackages:
    @vitejs/plugin-vue: ^4.0.0 => 4.2.1 
    vite: ^4.1.4 => 4.3.5
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitejs/vite/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitejs/vite/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitejs.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitejs/vite-plugin-vue/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a Vite issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to [vuejs/core](https://github.com/vuejs/core) instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitejs/vite-plugin-vue/discussions) or join our [Discord Chat Server](https://chat.vitejs.dev/).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sapphi-red** [comment, marked as answer]
esbuild and terser both have an option to do that but it has some caveats.
https://esbuild.github.io/api/#mangle-props

**Comment by arwinvdv** [comment]
@sapphi-red Thanks for the response! 

I tried in esbuild and terser both with this configuration in vite.config.js, but unfortunately no difference:
```
{ 
    build: {
      minify: 'terser',
      terserOptions: {
        mangle: {
          toplevel: true,
          module: true,
          properties: {
            regex: /^_/,
        },
      },
      compress: {
          drop_console: true,
          drop_debugger: true,
      },
    },
  },
}
```

And with esbuild:
```
{
  esbuild: {
    mangleProps: /_$/
  },
  build: {
    minify: 'esbuild'
  }
}

```

**Reply by sapphi-red** [reply]
You have to rename your properties/methods to match that regexes.
